src/flow_graph.py

```diff
@@ -1322,7 +1359,10 @@ def is_reducible(self) -> bool:
             for s in n.children():
                 if s.loop is not None and n in s.loop.backedges:
                     continue
-                incoming_edges[s].remove(n)
+                try:
+                    incoming_edges[s].remove(n)
+                except KeyError:
+                    break
```

Comment:
At first glance, I think the `break` should be `continue` (if it's OK to have missing edges) or `return False` (if it's not OK) -- otherwise, right now the behavior seems dependent on the order of the children in the list?

I think the fix here might actually be to iterate over `set(children)` instead. If `n` isn't in `incoming_edges[s]`, then I think that means `s.parents` was not constructed correctly? (Sorry that this might be a pre-existing bug)

```python
            for s in set(n.children()):
                if s.loop is not None and n in s.loop.backedges:
                    continue
                incoming_edges[s].remove(n)
```

Reply:
Does that mean `n.children()` might have repeats? That makes sense actually, and is probably why my extremely old version of this PR didn't need this fix (I changed `children()` to a `Set`.)

Reply:
Oh wait, I'm wrong, my bad! 

`n.children()` *shouldn't* have repeats: the implementations for that fn in `SwitchNode` & `ConditionalNode` explicitly deduplicate the children.

I can take a look and figure out what's going on here.